src/v/cluster/health_monitor_backend.cc

```diff
@@ -370,20 +370,23 @@ health_monitor_backend::collect_remote_node_health(model::node_id id) {
       });
 }
 
-result<node_health_report>
-map_reply_result(result<get_node_health_reply> reply) {
+result<node_health_report> map_reply_result(
+  model::node_id target_node_id, result<get_node_health_reply> reply) {
     if (!reply) {
         return {reply.error()};
     }
     if (!reply.value().report.has_value()) {
         return {reply.value().error};
     }
+    if (reply.value().report->id != target_node_id) {
```

Comment:
What's the point of rechecking it here, if we anyway have checked it on the server side in `service::do_collect_node_health_report`? I guess it'll only make sense for some corner cases when nodes run different versions of Redpanda?

Reply:
exactly